docs/guides/configuring-custom-entities.md

```diff
@@ -0,0 +1,169 @@
+# Configuring Custom Entities
+
+This is an **advanced** level guide for users using custom entities in Kong.
+Most users do need to use this feature.
+
+Kong has in-built extensibility with its plugin architecture.
+Plugins in Kong have a `config` property where users can store configuration
+for any custom plugin and this suffices in most use cases.
+In some use-cases, custom entities are created in Kong via plugins
+which are used to store additional configuration.
+This guide elaborates on how such custom entities can be used Kong Ingress
+Controller.
+
+> Note: All entities shipped with Kong are supported by Kong Ingress Controller
+out of the box. This guide applies only if you have a custom entity in your
+plugin. Custom plugins have first-class support in Kong Ingress Controller
+via the `KongPlugin` CRD. Please read [this](../setting-up-custom-plugins.md)
+guide for details.
+
+## Caveats
+
+- The feature discussed in this guide apply for DB-less deployments of Kong.
+  The feature is not supported for deployments where Kong is used with a
+  database or Kong is used in hybrid mode.
+  For these deployments, it is recommended to configure custom entities
+  directly using Kong's Admin API.
+- Custom entities which have a foreign relation with other core entities in Kong
+  are not supported. Only entities which can exist by themselves and then
+  be referenced via plugin configuration are supported.
+
+## Creating a JSON representation of the custom entity
+
+In this section, we will learn how to create a JSON representation of
+a custom entity.
+
+Suppose you have a custom entity in Kong with the following schema in Kong:
```

Comment:
```suggestion
Suppose you have a custom entity with the following schema in your plugin source:
```
"schema in Kong" also works, though is a bit less specific. Main thing is to not have "in Kong" repeated twice.